src/librpmem/rpmem_fip.c

```diff
@@ -943,29 +985,157 @@ rpmem_fip_fini_lanes_apm(struct rpmem_fip *fip)
 }
 
 /*
- * rpmem_fip_persist_raw -- (internal) perform persist operation using
- * READ after WRITE mechanism
+ * rpmem_fip_tx_inc -- (internal) increment number of elements in TX
+ */
+static inline void
+rpmem_fip_tx_inc(struct rpmem_fip_plane *lanep)
+{
+	++lanep->base.tx_elems;
+}
+
+/*
+ * rpmem_fip_tx_set_empty -- (internal) zero number of elements in TX
+ */
+static inline void
+rpmem_fip_tx_set_empty(struct rpmem_fip_plane *lanep)
+{
+	RPMEM_ASSERT(!lanep->base.tx_is_flushing);
+	lanep->base.tx_elems = 0;
+}
+
+/*
+ * rpmem_fip_tx_require_flush -- (internal) is TX almost full
+ */
+static inline int
+rpmem_fip_tx_require_flush(struct rpmem_fip *fip, struct rpmem_fip_plane *lanep)
+{
+	RPMEM_ASSERT(lanep->base.tx_elems < fip->fi->tx_attr->size);
+	return lanep->base.tx_elems + 1 == fip->fi->tx_attr->size;
+}
+
+/*
+ * rpmem_fip_tx_is_flushing -- (internal) is TX flush started
+ */
+static inline int
+rpmem_fip_tx_is_flushing(struct rpmem_fip_plane *lanep)
```

Comment:
I'm not sure if I understand what "flushing" means. It seems to be variable that indicates whether or not to call "rpmem_fip_tx_flush_wait" on the next operation.